### PR TITLE
docs: add foreach-loop example

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to #3950

### CODE REVIEW

**Comment by magnus-madsen** [comment]
I think we should have two examples that fit in the box on https://flix.dev/ i.e. they need to be a lot shorter.

**Comment by jaschdoc** [comment]
I'll make another one tonight :)

**Comment by jaschdoc** [comment]
Something with arrays and string builders

**Comment by JonathanStarup** [comment]
Maybe there should also be a quick section on Static and how to avoid it?

**Comment by magnus-madsen** [comment]
> Maybe there should also be a quick section on Static and how to avoid it?

I think in the docs not for flix.dev

The purpose of flix.dev is to show cool stuff.

**Comment by JonathanStarup** [comment]
> > Maybe there should also be a quick section on Static and how to avoid it?
> 
> I think in the docs not for flix.dev
> 
> The purpose of flix.dev is to show cool stuff.

I was just thinkink of a short note about regions so curious people know what to look for

**Comment by jaschdoc** [comment]
Oh so we still want to merge this? :jack_o_lantern:

File: examples/iterating-over-arrays-with-foreach.flix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/// Flix supports writing code in an imperative
+/// style if it is preferable to you.
+///
+/// foreach-loops are great for working with
+/// mutable data structures.
+///
+/// This example implements copying an array
+/// in an imperative style.
+///
+def copy!(in: { in :: Array[a, r1] }, out: { out :: Array[a, r2] }): Unit \ { Read(r1), Write(r2) } =
```
Comment (jaschdoc) [inline examples/iterating-over-arrays-with-foreach.flix:11]: Should we remove the records to reduce complexity?

**Comment by jaschdoc** [comment]
Added some more examples but I'm not sure how interesting they are. Any ideas?

**Comment by magnus-madsen** [comment]
I think the example should be short; so they fit in the box (or maybe 1.5 of the box).

I think they should show the features, so both a simple loop, use of pattern matching, use if nested-loops, use of two loops in one, and use of the if construct.

**Comment by magnus-madsen** [comment]
For example maybe something like:

```
def main(): Unit & Impure =
    use Array.toIterator;
    region r { 
    let xs = [1, 2, 3, 4] @ r;
    let ys = [1, 2, 3, 4] @ r;
    foreach (x <- toIterator(r, xs); 
             y <- toIterator(r, ys))
        println(x + y)
    }
```

with some comments thrown in.

**Comment by jaschdoc** [comment]
Okay I wasn't sure about the level of complexity I had to demonstrate the feature at. The above example should be doable :)

**Comment by jaschdoc** [comment]
What do you think of these examples (excluding `imperative-style-foreach-loops.flix`)?

**Comment by magnus-madsen** [comment]
Ill take a look, but we can let this brew while you work on the other for loop :)

**Comment by magnus-madsen** [comment]
Do you want to take another stab at these examples? I would just try to simplify the long one and then we can merge them in :)

**Comment by jaschdoc** [comment]
Hmm... the long example just explains the functionality of `foreach` so maybe we should just delete it, as it does not showcase an example but serves more as documentation of the feature (i.e. something we could put in the book).

File: examples/imperative-style-foreach-loops.flix [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+/// Flix supports writing code in an imperative
+/// style if it is preferable to you.
+///
+/// One of the features related to this is a
+/// foreach-loop.
+///
+/// In this example we explore how this loop
+/// works in Flix.
+///
+def main(): Unit & Impure =
```
Comment (magnus-madsen) [inline examples/imperative-style-foreach-loops.flix:11]: should be IO

Comment (jaschdoc) [reply examples/imperative-style-foreach-loops.flix:11]: Yes, but this is the one I'm suggesting we remove (that's why I didn't touch it) :)

**Comment by magnus-madsen** [comment]
I will merge this once CI completes, but think about what examples you would want for flix.dev and what you would want for doc.flix.dev.

**Pull request merged**